Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,271 @@
+    nth_rw 2 [← (h.out.1 ⟨σ, hσ⟩)]
+    exact AlgEquiv.restrictNormal_commutes σ L ⟨x, xL⟩
+  · have xL := IntermediateField.lift_le _ h
+    apply (IntermediateField.mem_lift (⟨x,xL⟩ : L)).mp at h
+    simp only [IntermediateField.mem_fixedField_iff, Subgroup.mem_map, forall_exists_index, and_imp,
+      forall_apply_eq_imp_iff₂] at h
+    simp only [IntermediateField.coe_inf, Set.mem_inter_iff, SetLike.mem_coe,
+      IntermediateField.mem_fixedField_iff, xL, and_true]
+    intro σ hσ
+    have : ((restrictNormalHom L σ) ⟨x, xL⟩).1 = x := by rw [h σ hσ]
+    nth_rw 2 [← this]
+    exact (AlgEquiv.restrictNormal_commutes σ L ⟨x, xL⟩).symm
+
+lemma fixingSubgroup_fixedField (H : ClosedSubgroup (K ≃ₐ[k] K)) [IsGalois k K] :
+    (IntermediateField.fixedField H).fixingSubgroup = H.1 := by
+  apply le_antisymm _ ((IntermediateField.le_iff_le H.toSubgroup
+    (IntermediateField.fixedField H.toSubgroup)).mp fun ⦃x⦄ a ↦ a)
+  intro σ hσ
+  by_contra h
+  have nhd : H.carrierᶜ ∈ nhds σ := H.isClosed'.isOpen_compl.mem_nhds h
+  rw [GroupFilterBasis.nhds_eq (x₀ := σ) (galGroupBasis k K)] at nhd
+  rcases nhd with ⟨b,⟨gp,⟨L,hL,eq'⟩,eq⟩,sub⟩
+  rw [← eq'] at eq
+  have sub : σ • b ⊆ H.carrierᶜ := Set.smul_set_subset_iff.mpr sub
+  have := hL.out
+  let L' : FiniteGaloisIntermediateField k K := {
+    normalClosure k L K with
+    finiteDimensional := normalClosure.is_finiteDimensional k L K
+    isGalois := IsGalois.normalClosure k L K }
+  have compl : σ • L'.1.fixingSubgroup.carrier ⊆ H.carrierᶜ :=
+    fun ⦃a⦄ d ↦ sub ((Set.set_smul_subset_set_smul_iff.mpr <| eq ▸ (fun σ h =>
+    (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mpr fun y hy => (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mp
+    h y (IntermediateField.le_normalClosure L hy))) d)
+  have fix : ∀ x ∈ IntermediateField.fixedField H.toSubgroup ⊓ ↑L', σ x = x :=
+    fun x hx ↦ ((mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mp hσ) x hx.1
+  rw [restrict_fixedField H.1 L'.1] at fix
+  have : (restrictNormalHom L') σ ∈ (Subgroup.map (restrictNormalHom L') H.1) := by
+    rw [← IntermediateField.fixingSubgroup_fixedField (Subgroup.map (restrictNormalHom L') H.1)]
+    apply (mem_fixingSubgroup_iff (L' ≃ₐ[k] L')).mpr
+    intro y hy
+    apply Subtype.val_injective
+    simp only [AlgEquiv.smul_def, restrictNormalHom_apply L'.1 σ y,
+      fix y.1 ((IntermediateField.mem_lift y).mpr hy)]
+  rcases this with ⟨h, mem, eq⟩
+  have : h ∈ σ • L'.1.fixingSubgroup.carrier := by
+    use σ⁻¹ * h
+    simp only [Subsemigroup.mem_carrier, Submonoid.mem_toSubsemigroup, Subgroup.mem_toSubmonoid,
+      smul_eq_mul, mul_inv_cancel_left, and_true]
+    apply (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mpr
+    intro y hy
+    simp only [AlgEquiv.smul_def, AlgEquiv.mul_apply]
+    have : ((restrictNormalHom L') h ⟨y,hy⟩).1 = ((restrictNormalHom L') σ ⟨y,hy⟩).1 := by rw [eq]
+    rw [restrictNormalHom_apply L'.1 h ⟨y, hy⟩, restrictNormalHom_apply L'.1 σ ⟨y, hy⟩] at this
+    simp only [this, ← AlgEquiv.mul_apply, inv_mul_cancel, one_apply]
```

Comment:
```suggestion
    have : ((restrictNormalHom L') h ⟨y,hy⟩).1 = ((restrictNormalHom L') σ ⟨y,hy⟩).1 := by rw [eq]
    rw [restrictNormalHom_apply L'.1 h ⟨y, hy⟩, restrictNormalHom_apply L'.1 σ ⟨y, hy⟩] at this
    simp only [AlgEquiv.smul_def]
```

Reply:
Does this realy work?

Reply:
It worked for me, have you tried? At any rate the answer is easy: if it works, go for it, if it does not (i.e. I made a mistake), forget it 😄

Reply:
I tried, even before this I once tried myself to reduce this and failed...

Reply:
Oh then forget it, I am sorry 😞